bundle/src/main/java/com/meltmedia/dropwizard/etcd/cluster/ClusterAssignmentService.java

```diff
@@ -222,10 +269,10 @@ public void startNodeAssignmentTask() {
                   ignoreException(() -> TimeUnit.SECONDS.sleep(1));
                 }
               }
-            } else if (processCount > maxProcessCount) {
-              for (String toUnassign : processes.get(thisNode.getId())) {
+            } else if ((currentNodeCount == 0 && processCount > 0) || ((processCount > maxProcessCount || !active) && unassignedProcessCount == 0)) {
```

Comment:
These conditions are getting complex, move them up above and give them clear names.